edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/RegistryController.cs

```diff
@@ -0,0 +1,693 @@
+                    await this.SendResponseAsync(HttpStatusCode.BadRequest, FormatErrorResponseMessage($"Invalid request auth chain {requestData.AuthChain}."));
+                    return;
+                }
+
+                if (!await this.AuthenticateAsync(actorDeviceId, Option.Some<string>(Constants.EdgeHubModuleId)))
+                {
+                    await this.SendResponseAsync(HttpStatusCode.Unauthorized);
+                    return;
+                }
+
+                // Check that the actor device is authorized to act OnBehalfOf the target
+                IEdgeHub edgeHub = await this.edgeHubGetter;
+                IDeviceScopeIdentitiesCache identitiesCache = edgeHub.GetDeviceScopeIdentitiesCache();
+                string targetId = $"{targetDeviceId}/{requestData.ModuleId}";
+                (bool authorized, string authChain) = await this.AuthorizeActorAsync(identitiesCache, actorDeviceId, Constants.EdgeHubModuleId, targetId);
+                if (!authorized)
+                {
+                    await this.SendResponseAsync(HttpStatusCode.Unauthorized);
+                    return;
+                }
+
+                Events.Authorized(nameof(this.DeleteModuleOnBehalfOfAsync), actorDeviceId, Constants.EdgeHubModuleId, targetId, authChain);
+                string edgeDeviceId = edgeHub.GetEdgeDeviceId();
+                RegistryApiHttpResult result = await this.apiClient.DeleteModuleAsync(
+                    edgeDeviceId,
+                    new DeleteModuleOnBehalfOfData($"{AuthChainHelpers.SkipFirstIdentityFromAuthChain(authChain)}", requestData.ModuleId));
+                await this.SendResponseAsync(result.StatusCode, result.JsonContent);
+                Events.CompleteRequest(nameof(this.DeleteModuleOnBehalfOfAsync), edgeDeviceId, targetId, requestData.AuthChain, result);
+            }
+            catch (Exception ex)
+            {
+                Events.InternalServerError(nameof(this.DeleteModuleOnBehalfOfAsync), ex);
+                await this.SendResponseAsync(HttpStatusCode.InternalServerError, FormatErrorResponseMessage(ex.ToString()));
+            }
+        }
+
+        async Task<bool> AuthenticateAsync(string deviceId, Option<string> moduleId)
+        {
+            IHttpRequestAuthenticator authenticator = await this.authenticatorGetter;
+            HttpAuthResult authResult = await authenticator.AuthenticateAsync(deviceId, moduleId, this.HttpContext);
+
+            if (authResult.Authenticated)
+            {
+                Events.Authenticated(deviceId, moduleId.GetOrElse(string.Empty));
+                return true;
+            }
+
+            Events.AuthenticateFail(deviceId, moduleId.GetOrElse(string.Empty));
+            return false;
+        }
+
+        async Task<(bool, string)> AuthorizeActorAsync(IDeviceScopeIdentitiesCache identitiesCache, string actorDeviceId, string actorModuleId, string targetId)
```

Comment:
>actorModuleId [](start = 129, length = 13)

not needed since you're always calling with $edgeHub as a constant string anyways